bridge/server/utils/configuration.ts

```diff
@@ -40,10 +229,170 @@ export function getConfiguration(options?: BridgeOption): BridgeConfiguration {
-      enabledComponents: loggingComponents,
+    destination: logDestination,
+    enabledComponents: loggingComponents,
+  };
+}
+
+function getAPI(options?: BridgeOption): APIConfig {
+  const _showToken = options?.api?.showToken ?? toBool(process.env[EnvVar.SHOW_API_TOKEN] ?? 'true');
+  const apiUrl = options?.api?.url ?? process.env[EnvVar.API_URL];
+  if (typeof apiUrl !== 'string') {
+    throw new Error('API_URL is not provided');
+  }
+  let apiToken = options?.api?.token ?? process.env[EnvVar.API_TOKEN];
+  if (typeof apiToken !== 'string') {
+    log.warning(_componentName, 'API_TOKEN was not provided. Fetching it from the K8s secrets via kubectl.');
+    apiToken =
+      Buffer.from(
+        execSync('kubectl get secret keptn-api-token -n keptn -ojsonpath={.data.keptn-api-token}').toString(),
+        'base64'
+      ).toString() || undefined;
+    if (typeof apiToken !== 'string') {
+      log.warning(_componentName, 'Could not fetch API_TOKEN from k8s secret.');
+    }
+  }
+
+  return {
+    showToken: _showToken,
+    url: apiUrl,
+    token: apiToken,
+  };
+}
+
+function getAuth(api: APIConfig, options?: BridgeOption): AuthConfig {
+  const authMsg =
+    options?.auth?.authMessage ??
+    process.env[EnvVar.AUTH_MSG] ??
+    `keptn auth --endpoint=${api.url} --api-token=${api.token}`;
+  const basicUser = process.env[EnvVar.BASIC_AUTH_USERNAME];
+  const basicPass = process.env[EnvVar.BASIC_AUTH_PASSWORD];
+  const requestLimit = toInt(process.env[EnvVar.REQUEST_TIME_LIMIT], 60) * 60 * 1000;
+  const requestWithinTime = toInt(process.env[EnvVar.REQUESTS_WITHIN_TIME], 10);
+  const cleanBucket = toInt(process.env[EnvVar.CLEAN_BUCKET_INTERVAL], 60) * 60 * 1000;
+
+  return {
+    authMessage: authMsg,
+    basicUsername: basicUser,
+    basicPassword: basicPass,
+    requestTimeLimitMs: requestLimit,
+    nRequestWithinTime: requestWithinTime,
+    cleanBucketIntervalMs: cleanBucket,
+  };
+}
+
+function getOAuth(options?: BridgeOption): OAuthConfig {
+  const logoutURL = process.env[EnvVar.OAUTH_ALLOWED_LOGOUT_URLS] ?? '';
+  let baseURL = options?.oauth?.baseURL ?? process.env[EnvVar.OAUTH_BASE_URL];
+  let clientID = options?.oauth?.clientID ?? process.env[EnvVar.OAUTH_CLIENT_ID];
+  const clientSecret = process.env[EnvVar.OAUTH_CLIENT_SECRET];
+  let discoveryURL = options?.oauth?.discoveryURL ?? process.env[EnvVar.OAUTH_DISCOVERY];
+  const enabled = options?.oauth?.enabled ?? toBool(process.env[EnvVar.OAUTH_ENABLED] ?? 'false');
+  const nameProperty = process.env[EnvVar.OAUTH_NAME_PROPERTY];
+  const scope = process.env[EnvVar.OAUTH_SCOPE] ?? '';
+  const secureCookie = toBool(process.env[EnvVar.SECURE_COOKIE] ?? 'false');
+  const timeout = toInt(process.env[EnvVar.SESSION_TIMEOUT_MIN], 60);
+  const proxyHops = toInt(process.env[EnvVar.TRUST_PROXY], 1);
+  const validation = toInt(process.env[EnvVar.SESSION_VALIDATING_TIMEOUT_MIN], 60);
+  const algo = process.env[EnvVar.OAUTH_ID_TOKEN_ALG] ?? 'RS256';
```

Comment:
I was reading the [doc of the lib](https://github.com/panva/node-openid-client/blob/main/docs/README.md#new-clientmetadata-jwks-options) and I thought it can handle itself. But as a second thought, extra good defaults are always better